app/src/main/kotlin/com/adesso/movee/scene/moviedetail/MovieDetailViewModel.kt

```diff
@@ -41,13 +36,8 @@ class MovieDetailViewModel @Inject constructor(
 
     fun fetchMovieCredits(id: Long) {
         if (_movieCredits.value == null) {
-            bgScope.launch {
-                val movieCreditsResult =
-                    fetchMovieCreditsUseCase.run(FetchMovieCreditsUseCase.Params(id))
-
-                onUIThread {
-                    movieCreditsResult.either(::handleFailure, ::postMovieCredits)
-                }
+            fetchMovieCreditsUseCase.invoke(viewModelScope, FetchMovieCreditsUseCase.Params(id)) {
+                it.either(::handleFailure, ::postMovieCredits)
```

Comment:
postMovieCredits() uses LiveData's setValue() function which must be called from main thread. Either we need to use postValue(), or switch context when calling postMovieCredits()